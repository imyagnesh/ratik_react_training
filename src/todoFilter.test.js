import React from 'react';
import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react';
import TodoFilter from './todoFilter';

const fnFilter = jest.fn();

const setup = () =>
  render(<TodoFilter onFilter={fnFilter} />);

describe('Todo Filter Component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component', () => {
    const { queryAllByRole } = setup();
    const buttons = queryAllByRole('button');
    expect(buttons.length).toBe(3);
    expect(buttons[0]).toHaveTextContent('All');
    expect(buttons[1]).toHaveTextContent('Pending');
    expect(buttons[2]).toHaveTextContent('Completed');
  });

  it('should click all button', () => {
    const { queryByText } = setup();
    const btnAll = queryByText('All');
    fireEvent.click(btnAll);
    expect(fnFilter).toBeCalledTimes(1);
    expect(fnFilter).toBeCalledWith('all');
  });

  it('should click pending button', () => {
    const { queryByText } = setup();
    const btnPending = queryByText('Pending');
    fireEvent.click(btnPending);
    expect(fnFilter).toBeCalledTimes(1);
    expect(fnFilter).toBeCalledWith('pending');
  });

  it('should click completed button', () => {
    const { queryByText } = setup();
    const btnCompleted = queryByText('Completed');
    fireEvent.click(btnCompleted);
    expect(fnFilter).toBeCalledTimes(1);
    expect(fnFilter).toBeCalledWith('completed');
  });

  it('should not rerender page', () => {
    const { rerender } = setup();
    rerender(<TodoFilter onFilter={fnFilter} />);
    const buttons = screen.queryAllByRole('button');
    expect(buttons.length).toBe(3);
  });
});
