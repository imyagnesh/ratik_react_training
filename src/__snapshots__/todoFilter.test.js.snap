// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Filter Component should match snapshot 1`] = `
<div>
  <button
    type="button"
  >
    All
  </button>
  <button
    type="button"
  >
    Pending
  </button>
  <button
    type="button"
  >
    Completed
  </button>
</div>
`;
